### PR TITLE
#768 Consistently using Type#getImportTypes() for determining imported types

### PR DESCRIPTION
**Pull request by gunnarmorling**
Hey @sjaakd and @agudian, another one related to imports. This one shoud improve it a fair bit.

### CODE REVIEW

**Comment by sjaakd** [comment]
@gunnarmorling : I'll have a look later this week...

File: processor/src/test/java/org/mapstruct/ap/test/imports/sourcewithextendsbound/SpaceshipMapper.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.mapstruct.ap.test.imports.sourcewithextendsbound.dto.SpaceshipDto;
+import org.mapstruct.ap.test.imports.sourcewithextendsbound.entity.Spaceship;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(uses = AstronautMapper.class)
+public interface SpaceshipMapper {
+
+    SpaceshipMapper INSTANCE = Mappers.getMapper( SpaceshipMapper.class );
+
+    SpaceshipDto spaceshiToDto(Spaceship spaceship);
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/imports/sourcewithextendsbound/SpaceshipMapper.java]: `spaceshiToDto` misses a `p`

**Comment by agudian** [comment]
@gunnarmorling looks good! If you add the missing p in the test, you can go ahead and push it to master... :+1:

**Comment by gunnarmorling** [comment]
Fixed the typo and merged. Thanks, A. "Eagle Eye" Gudian ;)

**Pull request closed without merging**